src/common/mutex_debug.h

```diff
@@ -93,6 +93,10 @@ class mutex_debug_impl : public mutex_debugging_base
   static constexpr bool recursive = Recursive;
 
   // Mutex concept is DefaultConstructible
+  mutex_debug_impl(bool bt = false)
+    : mutex_debugging_base("unnamed", bt) {
+    _init();
+  }
```

Comment:
is this change just motivated by convenience? i worry that making the name optional just makes the debug mutexes less useful in general

Reply:
yeah i'd rather require the names